### PR TITLE
Fix daily CI builds to publish tar package as artifacts

### PR DESCRIPTION
**Pull request by adityapatwardhan**
## PR Summary

Update the packaging script to always create the tar.gz packages.

## PR Context  

Daily builds for Linux and MacOS do not publish tar.gz package.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by PaulHigin — APPROVED** [review]

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -940,15 +940,15 @@ function New-MacOsDistributionPackage
     try
     {
         # productbuild is an xcode command line tool, and those tools are installed when you install brew
-        Start-NativeExecution -sb {productbuild --distribution $distributionXmlPath --resources $resourcesDir $newPackagePath}
+        Start-NativeExecution -sb {productbuild --distribution $distributionXmlPath --resources $resourcesDir $newPackagePath} > $null
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1]: There is a parameter that will suppress output unless there is an error.

File: tools/packaging/packaging.psm1 [hunk 2]
```diff
@@ -2234,7 +2234,8 @@ function New-NugetContentPackage
     $nupkgFile = "${nugetFolder}\${nuspecPackageName}-${packageRuntime}.${nugetSemanticVersion}.nupkg"
     if (Test-Path $nupkgFile)
     {
-        Get-ChildItem $nugetFolder\* | Select-Object -ExpandProperty FullName
+        # Get-ChildItem $nugetFolder\* | Select-Object -ExpandProperty FullName
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1]: remove commented code?

**Review by TravisEz13 — APPROVED** [review]

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (adityapatwardhan) [reply tools/packaging/packaging.psm1]: Fixed. Used `-VerboseOutputOnError`

File: tools/packaging/packaging.psm1 (hunk 2 above)
Comment (adityapatwardhan) [reply tools/packaging/packaging.psm1]: Sorry missed that. Removed.

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**